design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -60,16 +60,17 @@ type CustomerOrder {
     order_date: String! @doc("date when the order was placed")
     status: String! @doc("current status of the order")
     number: String! @doc("sequential order number")
-    items: [OrderItem]! @doc("collection of all the items purchased")
+    items: [OrderItemInterface]! @doc("collection of all the items purchased")
     total: OrderTotal! @doc("total amount details for the order")
     invoices: [Invoice]! @doc("invoice list for the order")
     credit_memos: [CreditMemo]! @doc("credit memo list for the order")
     shipments: [OrderShipment]! @doc("shipment list for the order")
     payment_methods: [PaymentMethod]! @doc("payment details for the order")
-    shipping_address: CustomerAddress! @doc("shipping address for the order")
+    shipping_address: CustomerAddress @doc("shipping address for the order")
     billing_address: CustomerAddress! @doc("billing address for the order")
-    carrier: String! @doc("shipping carrier for the order delivery")
-    method: String! @doc("shipping method for the order")
+    carrier: String @doc("shipping carrier for the order delivery")
+    method: String @doc("shipping method for the order")
+    comments: [CommentItem]! @doc("comments on the order")
```

Comment:
Let's not mark comments with `!`. Our latest vision is to not mark complex fields that may have their own resolver with `!` even if they are technically always present.
The reason is that it will allow for better exception handling (exception will not bubble up to the nearest nullable field), and we can always add `!` in the future to the output types (cannot remove though). 

Please check all complex fields in output types across this new schema.